### ISSUE TITLE
Upload terminology

### ISSUE THREAD

**Issue by lrasmus** [microsoft/fhir-server#810]
**Preamble**
This is presented as a user story, but really I'm wondering if this is already possible (and if so, how).  This stems from an issue I closed since it made a poor assumption about how this functionality would be handled (see #809).

**User story**
As a developer setting up a local instance of fhir-server, I would like to be able to upload an external terminology so that it may be used by other resources.  This will streamline the setup of fhir-server when terminologies like LOINC and SNOMED-CT are needed.

**Acceptance criteria**
1. There should be a documented process by which I can upload terminologies such as LOINC and SNOMED-CT.  Similar in spirit to how hapi-fhir-cli handles it: `hapi-fhir-cli upload-terminology -v r4 -t http://localhost:8080/fhir/ -u http://snomed.info/sct -d SnomedCT_USEditionRF2_PRODUCTION_20190301T120000Z.zip`

**Comment by hansenms** [comment]
We don't have this capability yet. We are considering how we would want to support this. It is related to profile validation and so on. I don't have an estimate or timeline yet, but this is something we are interested in.

**Comment by lrasmus** [comment]
@hansenms - happy to contribute to this, if there's a general approach you'd like to see taken for its implementation.

**Comment by hansenms** [comment]
We certainly appreciate the offer for help an once we get this broken down a bit more, I am sure there could be things to pick up, but there are quite a few moving parts. Some of it is related to validation (profiles, valuesets, etc). That work has started if you follow the PRs. We then need the ability to allow uploads of value sets profiles, etc. The consensus in the community is that this should be FHIR packages (npm packages) and the upload terminology functionality that this started with. We have to understand better how this will work. Then there is the actually functionality that terminology services could provide...like the ability to search `:above` and `:below` codes, etc. That is another big topic where we may actually want to delegate the functionality to an external terminology server.

So bottom line, this is a big sprawling topic, and there are no concrete work items that can easily be handed out. We do appreciate the offer and I would encourage to take a look at some of the up for grabs items.

**Comment by MattCordell** [comment]
I was going to ask what terminology capabilities this fhir-server will has, and found this question. Seems like the answer is - none at the moment?

I agree with everything above, including the original issue. Each codesystem is going to need a dedicated process to import. At least SNOMED CT and LOINC will need their own, plus a generic "code+term" text import.

**Comment by hansenms** [comment]
@MattCordell , you are correct we do not support terminology services capabilities yet.

I don't think we will be supporting some custom upload functionality like the one provided by the hapi client. We plan on supporting FHIR packages (https://github.com/FHIR/packages), which seems to be the community consensus on where this should be going.

**Comment by CaitlinV39** [comment]
We have this documented as an item that customers are interested in but it isn't currently prioritized and we don't have a full plan on how we would approach terminology services. For now, I'm going to close this as won't fix
